### DISCUSSION TITLE
How get config from _config.yml in scripts?

### DISCUSSION THREAD

**Discussion by gjssss** [hexojs/hexo#4964] [Question / Answer] [answered]
When I use ` hexo.theme.config `, I got a empty object
and use ` hexo.config` can only got globle config `_config.yml`

How can I get Theme Config `themes/test/_config.yml`  in `themes/test/scripts/test.js`!

**Comment by yoshinorin** [comment, marked as answer]
@gjssss

In hexo `6.x.x` you can get the theme config values like the below way.

```js
// themes/<your_theme>/scripts/example.js
function themeConfigConsoleLog() {
  console.log(this.config.theme_config);
}

// Please see: https://hexo.io/api/filter
hexo.extend.filter.register('after_init', themeConfigConsoleLog)

// Also you can use it in a template engin (e.g: EJS)
// https://hexo.io/docs/helpers
hexo.extend.helper.register('<helper_name>', themeConfigConsoleLog)
```

If you have a `_theme.<your_theme>.yml`.

```yaml
// _theme.<your_theme>.yml
title: test
```

You can see theme config when start a server.

```sh
$ hexo server

INFO  Validating config
{ title: 'test' }
INFO  Start processing
```

**Reply by gjssss** [reply]
Thank You!
But it is not work when I have a config file `_theme.test.yaml` in the root
So I rename the file as `_config.test.yml`, Everything is OK now!

**Reply by yoshinorin** [reply]
> But it is not work when I have a config file _theme.test.yaml in the root
> So I rename the file as _config.test.yml, Everything is OK now!

Thank you. I fixed my comment :)
